typescript/sdk/src/token/checker.ts

```diff
@@ -0,0 +1,90 @@
+import { BigNumber } from 'ethers';
+
+import { ERC20, ERC20__factory } from '@hyperlane-xyz/core';
+
+import { TokenMismatchViolation } from '../deploy/types';
+import { HyperlaneRouterChecker } from '../router/HyperlaneRouterChecker';
+import { ChainName } from '../types';
+
+import { HypERC20App } from './app';
+import {
+  ERC20RouterConfig,
+  HypERC20Config,
+  TokenMetadata,
+  isCollateralConfig,
+  isNativeConfig,
+  isSyntheticConfig,
+} from './config';
+import { HypERC20Factories } from './contracts';
+
+export class HypERC20Checker extends HyperlaneRouterChecker<
+  HypERC20Factories,
+  HypERC20App,
+  ERC20RouterConfig
+> {
+  async checkChain(chain: ChainName): Promise<void> {
+    await super.checkChain(chain);
+    await this.checkToken(chain);
+  }
+
+  async checkToken(chain: ChainName): Promise<void> {
+    const checkERC20 = async (
+      token: ERC20,
+      config: HypERC20Config | boolean,
```

Comment:
why `| boolean` ?

Reply:
I'm overloading this for the "if the ERC20 exists here" check

Reply:
talked to yorke and simplified it